geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/builder/FeatureTypeBuilder.scala

```diff
@@ -0,0 +1,119 @@
+package org.locationtech.geomesa.utils.geotools.builder
+
+import java.util.UUID
+
+import org.geotools.feature.`type`.AttributeDescriptorImpl
+import org.geotools.feature.{AttributeImpl, ComplexAttributeImpl, FeatureImpl, NameImpl}
+import org.geotools.filter.identity.FeatureIdImpl
+import org.geotools.gml3.GMLSchema
+import org.opengis.feature.`type`._
+import org.opengis.feature.{ComplexAttribute, Feature, Property}
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+
+object FeatureTypeBuilder {

```

Comment:
is this only for complex types? we have the SftBuilder for simple types so i assume it is...if so lets name it ComplexTypeBuilder or something like that